clients/src/test/java/org/apache/kafka/clients/ClientUtilsTest.java

```diff
@@ -26,30 +26,58 @@
 
 public class ClientUtilsTest {
 
+
     @Test
     public void testParseAndValidateAddresses() {
-        check("127.0.0.1:8000");
-        check("mydomain.com:8080");
-        check("[::1]:8000");
-        check("[2001:db8:85a3:8d3:1319:8a2e:370:7348]:1234", "mydomain.com:10000");
-        List<InetSocketAddress> validatedAddresses = check("some.invalid.hostname.foo.bar.local:9999", "mydomain.com:10000");
+        checkWithoutLookup("127.0.0.1:8000");
+        checkWithoutLookup("mydomain.com:8080");
+        checkWithoutLookup("[::1]:8000");
+        checkWithoutLookup("[2001:db8:85a3:8d3:1319:8a2e:370:7348]:1234", "mydomain.com:10000");
```

Comment:
Some day, mydomain.com will be changed and all of this will break.  Additionally, this prevents people from running the unit tests without network connectivity.  Our unit tests shouldn't have this dependency on an external network resource.

We should be able to figure out what the IP address is for localhost and go from there.

If we want to validate a non-localhost address as well, we can use the standard "example.com"  (See https://en.wikipedia.org/wiki/Example.com )

Reply:
Good point. I was merely following what was already in place.
Would we want to address this now or could the feature be merged as is and I can take a stab at having something better for this test next week?

Reply:
updated the tests.